src/devices/cpu/ht1130/ht1130.h

```diff
@@ -0,0 +1,130 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#ifndef MAME_CPU_HT1130_HT1130_H
+#define MAME_CPU_HT1130_HT1130_H
+
+#pragma once
+
+
+class ht1130_device : public cpu_device
+{
+public:
+	ht1130_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	enum {
+		HT1130_PC,
+		HT1130_R0,
+		HT1130_R1,
+		HT1130_R2,
+		HT1130_R3,
+		HT1130_R4,
+		HT1130_ACC,
+		HT1130_TIMER_EN,
+		HT1130_TIMER,
+	};
+
+	auto pm_in_cb() { return m_port_in_pm.bind(); }
+	auto ps_in_cb() { return m_port_in_ps.bind(); }
+	auto pp_in_cb() { return m_port_in_pp.bind(); }
+
+	auto pa_out_cb() { return m_port_out_pa.bind(); }
+
+	auto display_data_out_cb() { return m_display_data_out.bind(); }
+
+	void wake() // can be configured to wake from certain inputs as a mask option?
+	{
+		m_inhalt = 0;
+	}
```

Comment:
Probably better to put this through a `synchronize` barrier to avoid time travel effects on the odd chance it gets used in a system with multiple diexec implementations.